test/features/courses/ui/courses_cubit_test.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:bloc_test/bloc_test.dart';
+import 'package:flutter_learn_app/features/courses/data/courses_repository.dart';
+import 'package:flutter_learn_app/features/courses/domain/models/course.dart';
+import 'package:flutter_learn_app/features/courses/ui/cubit/courses_cubit.dart';
+import 'package:flutter_learn_app/features/courses/ui/cubit/courses_state.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'courses_cubit_test.mocks.dart';
+
+@GenerateMocks([], customMocks: [MockSpec<CoursesRepository>()])
+void main() {
+  late CoursesCubit coursesCubit;
+  late CoursesRepository coursesRepository;
+
+  setUp(() {
+    coursesRepository = MockCoursesRepository();
+    coursesCubit = CoursesCubit(coursesRepository);
+  });
+
+  blocTest(
```

Comment:
add more tests
1. how about when coursesRepository.getAllCourses() returns empty list
2. what if coursesRepository.getAllCourses() throws exception?
3. tests for cubit.init() method?

Reply:
fixed